From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Thu, 19 Mar 2015 16:44:42 +0800
Subject: [PATCH] {NOT UPSTREAMABLE} arch/x86/xen/trace: force building without
 -Werror

because of a non-silentable warning about a comparison to function pointer
---
 arch/x86/xen/Makefile | 4 ++++
 init/Makefile         | 6 ++++++
 2 files changed, 10 insertions(+)

diff --git a/arch/x86/xen/Makefile b/arch/x86/xen/Makefile
index 4953260e281c..8e2db791d1b8 100644
--- a/arch/x86/xen/Makefile
+++ b/arch/x86/xen/Makefile
@@ -11,6 +11,10 @@ endif
 CFLAGS_enlighten_pv.o		:= -fno-stack-protector
 CFLAGS_mmu_pv.o			:= -fno-stack-protector
 
+ifdef CONFIG_CC_IS_CLANG
+CFLAGS_trace.o := -Wno-error
+endif
+
 obj-y				+= enlighten.o
 obj-y				+= mmu.o
 obj-y				+= time.o
diff --git a/init/Makefile b/init/Makefile
index 2846113677ee..1f438d9392e4 100644
--- a/init/Makefile
+++ b/init/Makefile
@@ -15,6 +15,12 @@ obj-$(CONFIG_GENERIC_CALIBRATE_DELAY) += calibrate.o
 
 obj-y                          += init_task.o
 
+# clang fails with "error: ordered comparison of function pointers ('initcall_t' (aka 'int (*)(void)') and 'initcall_t') [-Werror]
+# and it is not possible to work around easily
+ifdef CONFIG_CC_IS_CLANG
+CFLAGS_main.o += -Wno-error
+endif
+
 mounts-y			:= do_mounts.o
 mounts-$(CONFIG_BLK_DEV_RAM)	+= do_mounts_rd.o
 mounts-$(CONFIG_BLK_DEV_INITRD)	+= do_mounts_initrd.o
-- 
